Implement admin authorization

Mister-Corn opened the pull request:
_Description to be added_.

Latest commit hasn't been tested. Will test tomorrow. Current implementation needs to have `localhost:5000/api/admins/redirect` whitelisted in the Google OAuth2 settings, though I think it'll still work if we have it redirect to `localhost:5000/api/customers/redirect` like before. We'll see.

TODO:
* [x] ~Test if setting `admin` property in `request` object does anything~. It doesn't.
* [x] Test if admin information properly gets written to admin table.
* [ ] Have `checkToken` check `admin` property on token and authorize accordingly.

Mister-Corn committed:
Implement admin authorization

NandoTheessen commented on auth/google.js:
```diff
@@ -17,19 +13,21 @@ const passportConfig = {
+            let idPropName = request.admin ? 'admin_id' : 'cust_id';
+            return { [idPropName]: id, email: email.value}
         });
+        const group = request.admin ? 'admins' : 'customers';
```
Good idea to implement this way! 
what do you think about just adding a flag to the request body from the front end to do this?

Mister-Corn committed:
Implement different strategies for customers/admins

Other:
* Added morgan dependency for debugging/logging
* Moderate structural changes to better setup the different strategies.

Should be noted this approach here is kind of for fun: there are easier implementations than this.

Mister-Corn committed:
Test admin/customer registration and modify accordingly

Mister-Corn commented on ron.env:
```diff
@@ -0,0 +1,6 @@
+JWT_SECRET=0CRG3aprOR02H9AZkNruGcbgNxQSTXIbms0CHK03BIZ8mRGcHIlrkCdUeH9GheH
+GOOGLE_ID=228925631597-ha4a48qab9iod3ajhop0c9644fh7kdq8.apps.googleusercontent.com
+GOOGLE_SECRET=y1Cqtn3gabLvW10klTKQbxh1
+REACT_REDIRECT=http://localhost:5000/
```
herp derp derp

NandoTheessen committed:
Resolve merging conflicts
